ArcGISRuntimeSDKQt_CppSamples/LocalServer/LocalServerFeatureLayer/LocalServerFeatureLayer.cpp

```diff
@@ -60,37 +61,73 @@ void LocalServerFeatureLayer::componentComplete()
   // Set map to map view
   m_mapView->setMap(m_map);
 
-  QString dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data";
+  // Check for ArcGIS Pro map package files
+  QString fileName = "PointsofInterest.mpkx";
+  QString dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpkx/" + fileName;
+
+  // Check to see if map package exists
+  if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+  {
+    qDebug() << "ArcGIS Pro .mpkx file not found at" << dataPath << "\nChecking for .mpk file";
+    fileName = "PointsofInterest.mpk";
+    dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/" + fileName;
+    if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+      qDebug() << "File:" << dataPath << "not found";
+    else
+      qDebug() << "Using .mpk file from" << dataPath;
+  }
 
   // create a feature service
-  m_localFeatureService = new LocalFeatureService(dataPath + "/mpk/PointsofInterest.mpk", this);
+  m_localFeatureService = new LocalFeatureService(dataPath, this);
 
   if (LocalServer::instance()->isInstallValid())
   {
     connectSignals();
     if (LocalServer::status() == LocalServerStatus::Started)
-      m_localFeatureService->start();
-    else
+    {
+      qDebug() << "Local server was already running";
+
+      // The local server can already be running if it was started in a different application.
+      // We start up the local feature service here because there will be no status change signal to otherwise trigger it.
+
+      if (m_localFeatureService->status() != LocalServerStatus::Started || m_localFeatureService->status() != LocalServerStatus::Starting)
+      {
+        m_localFeatureService->start();
+      }
+    }
+    else if (LocalServer::status() != LocalServerStatus::Started || LocalServerStatus() != LocalServerStatus::Starting)
+    {
       LocalServer::start();
+    }
```

Comment:
That's a good idea, I'll make that change

Reply:
1. The else if wouldn't have triggered if the `LocalServer` was `Stopped`, `Stopping`, or `Failed`. I'm going to change that so the else if triggers to `LocalServer::start()` if the server status is not `Starting`.
2. I'm opting not to call `LocalServer::start()` if the LocalServer status is `Started` or `Starting` because I think that calculating an if statement is much faster than calling a function, regardless of the effect. @jared-2016 do you agree?
3. `LocalServerStatus()`, for all intents and purposes, seems to be identical to the `LocalServer::status()` function. I changed everything to the latter for consistency.